src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.c

```diff
@@ -1550,6 +1604,13 @@ static void lb_on_response_received_locked(grpc_exec_ctx *exec_ctx, void *arg,
     if (!glb_policy->seen_initial_response &&
         (response = grpc_grpclb_initial_response_parse(response_slice)) !=
             NULL) {
+      /* Don't use fallback any more. */
```

Comment:
Yes, good catch -- this should definitely be moved.  I would put it on line 1675, right after setting `glb_policy->serverlist` but before calling `rr_handover_locked()`.

The serverlist not being included in the initial response is just how the load balancer protocol is defined.  The initial response message contains different information than subsequent messages, as you can see here:

https://github.com/grpc/grpc/blob/master/src/proto/grpc/lb/v1/load_balancer.proto#L105